apps/glusterfs/app_device.go

```diff
@@ -322,35 +322,31 @@ func (a *App) DeviceSetState(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, "request unable to be parsed", 422)
 		return
 	}
+	logger.Info("entered set state state is %+v, device is %v", msg, id)
 
 	// Set state
-	err = a.db.Update(func(tx *bolt.Tx) error {
-		device, err := NewDeviceEntryFromId(tx, id)
+	a.asyncManager.AsyncHttpRedirectFunc(w, r, func() (string, error) {
+		err = a.db.Update(func(tx *bolt.Tx) error {
+			device, err := NewDeviceEntryFromId(tx, id)
+			if err != nil {
+				logger.LogError("Did not find device requested %v", id)
+				return err
+			}
+
+			// Set state
+			err = device.SetState(tx, a.executor, a.allocator, msg.State)
```

Comment:
Right... Using locks reduces parallelism. But the parallelism is also problematic. @raghavendra-talur can you explain in more detail, what problem you are solving with the introduction of the locks?